File: app/src/main/java/com/dhl/wanandroid/activity/WebActivity.java
package com.dhl.wanandroid.activity;

import android.app.Activity;
import android.content.Intent;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;

import androidx.appcompat.widget.Toolbar;

import android.text.Html;
import android.text.TextUtils;
import android.view.KeyEvent;
import android.view.View;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.RelativeLayout;
import android.widget.TextView;

import com.dhl.wanandroid.R;
import com.just.agentweb.AgentWeb;

/**
 * H5 页面
 * @author dhl
 */
public class WebActivity extends AppCompatActivity {


    private static final String TAG = "WebActivity";
    private AgentWeb mAgentWeb;
    private LinearLayout linearLayout;

    /**
     * 返回Lay
     */

    private RelativeLayout back_lay;
    private Toolbar toolbar;

    private TextView toolbar_title;

    private String title;

    private String url;

    private ImageView iv_back;

    public static void startActivity(Activity activity, String title, String url) {
        Intent intent = new Intent(activity, WebActivity.class);
        intent.putExtra("title", title);
        intent.putExtra("webUrl", url);
        activity.startActivity(intent);
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_web);
        initView();
        getIntentData();
        setWebView();
    }

    private void initView() {
        back_lay = findViewById(R.id.back_lay);
        linearLayout = findViewById(R.id.container);
        toolbar = findViewById(R.id.tool_bar);
        toolbar_title = findViewById(R.id.toolbar_title);
        iv_back = findViewById(R.id.iv_back);
        iv_back.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                if (!mAgentWeb.back()) {
                    finish();
                }
            }
        });
        back_lay.setVisibility(View.VISIBLE);
    }

    private void getIntentData() {
        Intent intent = getIntent();
        if (intent != null) {
            title = intent.getStringExtra("title");
            url = intent.getStringExtra("webUrl");
            toolbar_title.setText(Html.fromHtml(title));

        }

    }

    private void setWebView() {
        mAgentWeb = AgentWeb.with(this)
                .setAgentWebParent(linearLayout, new LinearLayout.LayoutParams(-1, -1))
                .useDefaultIndicator()
                .createAgentWeb()
                .ready()
                .go(url);
    }

    @Override
    public boolean onKeyDown(int keyCode, KeyEvent event) {
        if (mAgentWeb != null && mAgentWeb.handleKeyEvent(keyCode, event)) {
            return true;
        }

        return super.onKeyDown(keyCode, event);
    }
}

File: app/src/main/java/com/dhl/wanandroid/activity/WebActivity.kt
package com.dhl.wanandroid.activity

import android.app.Activity
import android.content.Intent
import android.os.Bundle
import android.text.Html
import android.view.KeyEvent
import android.widget.LinearLayout
import androidx.appcompat.widget.Toolbar
import com.dhl.wanandroid.R
import com.just.agentweb.AgentWeb

/**
 * H5 页面
 * @author dhl
 */
class WebActivity : BasicActivity() {
    private var mAgentWeb: AgentWeb? = null

    private var title: String? = null
    private var url: String? = null

    /**
     * 返回Lay
     */
    private val toolbar: Toolbar by lazy {
        findViewById(R.id.tool_bar)
    }

    private val linearLayout: LinearLayout by lazy {
        findViewById(R.id.container)
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_web)
        getIntentData()
        setWebView()
    }


    private fun getIntentData() {

        if (intent != null) {
            title = intent.getStringExtra("title")
            url = intent.getStringExtra("webUrl")
            toolbar!!.title = Html.fromHtml(title)
        }
        setSupportActionBar(toolbar)
        toolbar.run {
            setNavigationIcon(R.mipmap.back)
            setNavigationOnClickListener {
                finish()
            }
        }


    }


    private fun setWebView() {
        mAgentWeb = AgentWeb.with(this)
            .setAgentWebParent(linearLayout, LinearLayout.LayoutParams(-1, -1))
            .useDefaultIndicator()
            .createAgentWeb()
            .ready()
            .go(url)
    }

    override fun onKeyDown(keyCode: Int, event: KeyEvent): Boolean {
        return if (mAgentWeb != null && mAgentWeb!!.handleKeyEvent(keyCode, event)) {
            true
        } else super.onKeyDown(keyCode, event)
    }

    companion object {
        private const val TAG = "WebActivity"

        @JvmStatic
        fun startActivity(activity: Activity, title: String?, url: String?) {
            val intent = Intent(activity, WebActivity::class.java)
            intent.putExtra("title", title)
            intent.putExtra("webUrl", url)
            activity.startActivity(intent)
        }
    }
}